docs/developer-docs/build/using-an-agent.mdx

```diff
@@ -0,0 +1,56 @@
+import Tabs from '@theme/Tabs';
+import TabItem from '@theme/TabItem';
+import CodeBlock from '@theme/CodeBlock';
+
+# Using an Agent
+
+If you have a canister running, either on your local machine or live on the Internet Computer, you will have two main ways to interact with your canister smart contract. You can talk to the canister using the v2 API using an `agent` that follows the interface specification, or you can use the canister's http interface.
+
+## Calling a Canister
+
+In the Internet Computer ecosystem, a library that is used to make calls to the IC public interface is called an agent. An agent has a few key responsibilities, which make it convenient to work with in your language of choice. 
+
+A list of agents is available at [other languages -> agents](./languages/other-languages/other-languages-intro#interfacing-with-canisters-agents).
+
+### 1. Structuring Data
+
+A `call` to the Internet Computer can take two common forms - an `update` or a `query`. The `agent` submits a POST request to `/api/v2/canister/<effective_canister_id>/call`, and includes the following components: 
+* `request_type`
+* Authentication
+    * `sender`, `nonce`, and `ingress_expiry`
+* `canister_id`
+* `method_name`
+* `arg` - the rest of the payload
+
+By knowing the Candid interface of the canister, the `agent` will assemble the `"arg"` by `cbor`-encoding the data from the client application, and ensuring it matches the Candid interface for the method it will be calling. That encoded data will be turned into a buffer.
```

Comment:
IIUC, we `cbor` encode the whole authentication envelope, not only on the `arg`.